docs/mkdocs/docs/integration/migration_guide.md

```diff
@@ -0,0 +1,260 @@
+# Migration Guide
+
+This page collects some guidelines on how to future-proof your code for future versions of this library.
+
+## Replace deprecated functions
+
+The following functions have been deprecated and will be removed in the next major version (i.e., 4.0.0). All
+deprecations are annotated with
+[`HEDLEY_DEPRECATED_FOR`](https://nemequ.github.io/hedley/api-reference.html#HEDLEY_DEPRECATED_FOR) to report which
+function to use instead.
+
+#### Parsing
+
+- Function `friend std::istream& operator<<(basic_json&, std::istream&)` is deprecated since 3.0.0. Please use
+  [`friend std::istream&  operator>>(std::istream&, basic_json&)`](../api/operator_gtgt.md) instead.
+  
+    === "Deprecated"
+    
+        ```cpp
+        nlohmann::json j;
+        std::stringstream ss("[1,2,3]");
+        j << ss;
+        ```
+    
+    === "Future-proof"
+
+        ```cpp
+        nlohmann::json j;
+        std::stringstream ss("[1,2,3]");
+        ss >> j;
+        ```
+
+- Passing iterator pairs or pointer/length pairs to parsing functions ([`parse`](../api/basic_json/parse.md),
+  [`accept`](../api/basic_json/accept.md), [`sax_parse`](../api/basic_json/sax_parse.md),
+  [`from_cbor`](../api/basic_json/from_cbor.md), [`from_msgpack`](../api/basic_json/from_msgpack.md),
+  [`from_ubjson`](../api/basic_json/from_ubjson.md), and [`from_bson`](../api/basic_json/from_bson.md) via initializer
+  lists is deprecated since 3.8.0. Instead, pass two iterators; for instance, call `from_cbor(ptr, ptr+len)` instead of
+  `from_cbor({ptr, len})`.
+
+    === "Deprecated"
+  
+          ```cpp
+          const char* s = "[1,2,3]";
+          bool ok = nlohmann::json::accept({s, s + std::strlen(s)});
+          ```
+  
+    === "Future-proof"
+  
+          ```cpp
+          const char* s = "[1,2,3]";
+          bool ok = nlohmann::json::accept(s, s + std::strlen(s));
+          ```
+
+#### JSON Pointers
+
+- Comparing JSON Pointers with strings via [`operator==`](../api/json_pointer/operator_eq.md) and
+  [`operator!=`](../api/json_pointer/operator_ne.md) is deprecated since 3.11.2. To compare a
+  [`json_pointer`](../api/json_pointer/index.md) `p` with a string `s`, convert `s` to a `json_pointer` first and use
+  [`json_pointer::operator==`](../api/json_pointer/operator_eq.md) or
+  [`json_pointer::operator!=`](../api/json_pointer/operator_ne.md).
```

Comment:
```suggestion
- Comparing JSON pointers with strings using the string overload of [`operator==`](../api/json_pointer/operator_eq.md)
  and [`operator!=`](../api/json_pointer/operator_ne.md) is deprecated since 3.11.2. To compare a
  [`json_pointer`](../api/json_pointer/index.md) `p` with a string `s`, convert `s` to a `json_pointer` first, then
  compare the two JSON pointers.
```

The C++20 change is an implementation detail. It has no bearing on the types involved in the comparison.